gateway/internal/context/context.go

```diff
@@ -0,0 +1,19 @@
+package context
+
+import (
+	"context"
+
+	gwtypes "github.com/rudderlabs/rudder-server/gateway/internal/types"
+)
+
+// GetRequestTypeFromCtx : get request type from context
+func GetRequestTypeFromCtx(ctx context.Context) (string, bool) {
+	reqType, ok := ctx.Value(gwtypes.CtxParamCallType).(string)
+	return reqType, ok
+}
+
+// GetAuthRequestFromCtx : get auth request from context
+func GetAuthRequestFromCtx(ctx context.Context) (*gwtypes.AuthRequestContext, bool) {
+	authReqCtx, ok := ctx.Value(gwtypes.CtxParamAuthRequestContext).(*gwtypes.AuthRequestContext)
+	return authReqCtx, ok
+}
```

Comment:
Shouldn't these be used everywhere now that they exist? What's the point of returning the second value (bool) if nobody cares about it in practice? Would it be more sincere to rename these functions to `MustGetXxx` and panic if casting fails?

Reply:
added handling for bool, we will fail request with ISE if we are not able to get these from context, As we don't want our application to panic in such cases.